Assignment 5.Rmd

```diff
@@ -40,42 +45,77 @@ We want to see if we can build a decision tree to help teachers decide which stu
 ```{r}
-
+hist(D1$score)
 ```
 
 #Create a categorical outcome variable based on student score to advise the teacher using an "ifelse" statement
+
+It seems that the distribution of score is "tri-modal". That is, there are 3 groups of students gathered around 3 different score values. We will reccoment "advice" for those students in the middle range, with seems to be bewwwn .3 and .6.
+
 ```{r}
-D1$advice <- 
+D1$advice <- as.factor(ifelse(.3 <D1$score & D1$score <= .6, 1, 0))
 ```
 
 #Build a decision tree that predicts "advice" based on how many problems students have answered before, the percentage of those problems they got correct and how many hints they required
 ```{r}
-score_ctree <- 
+score_ctree <- rpart(advice ~ prior_prob_count + prior_percent_correct + hints, method="class", data=D1)
+  
 ```
 
 #Plot tree
 ```{r}
-
+fancyRpartPlot(score_ctree, caption = NULL)
 ```
 
 Please interpret the tree, which two behaviors do you think the teacher should most closely pay attemtion to?
 
 #Test Tree
 Upload the data "intelligent_tutor_new.csv". This is a data set of a differnt sample of students doing the same problems in the same system. We can use the tree we built for the previous data set to try to predict the "advice" we should give the teacher about these new students. 
 
+ANSWER:
+
+1) Students that ask for any hints. The first behavior teachers should pay attention to is students that ask for any hints. This category includes 42% of the population, and among these, 48% should receive advice.
+
+2) Students whose percentage of prior correct answers is below .6 (among students that asked for hints).Among those students that asked for hints, paying attention to their prior correct answer percentage provides adittional information for teachers about which students are likely to benefit from receiving advice. Among students with prior correct answer rate below .6 AND who have asked for at least one hint, 60% should receive advice. This "dual" category ("at leas 1 hint" + "correct answer rate below .6") includes 37.7% of the total students who should receive advice ( = .22*.6/.35)  
+
 ```{r}
 #Upload new data
 
-D2 <- 
+D2 <- read.csv("~/R/HUDK/assignment5/intelligent_tutor_new.csv")
+D2$advice <- ifelse(.3 <D2$score & D2$score <= .6, 1, 0)
 
 #Generate predicted advice using the predict() command for new students based on tree generated from old students
 
-D2$prediction <- 
-
+D2$prediction <- predict(score_ctree, D2, type = "class")
+  
 ``` 
+
 ## Part III
+
 Compare the predicted advice with the actual advice that these students recieved. What is the difference between the observed and predicted results?
 
+```{r}
+#Upload new data
+
+D2 <- read.csv("~/R/HUDK/assignment5/intelligent_tutor_new.csv")
+D2$advice <- ifelse(D2$score == 1, 0, 1) # Anna told me that for this variable, students coded as "1" should not receive advice, while students coded "0" should (I find this actually very weired since everyone in the dataset has score =1)
```

Comment:
I agree, for the pure sake of value-label, that "advice = 1" means receive advice, and "advice=0" means no advice. But I am confused about the two different ways you use to define the variable "advice". I would feel the first one `D2$advice <- ifelse(.3 <D2$score & D2$score <= .6, 1, 0) `makes better sense for me, although the results are exactly the same since the score in D2 is a distinguishable variable (which is weird due to the dataset itself).